docs/Server.md

```diff
@@ -473,6 +473,19 @@ const fastify = require('fastify')({
 })
 ```
 
+<a name="rewrite-url"></a>
+### `rewriteUrl`
+
+Set a callback function that allows rewriting urls. 
```

Comment:
```suggestion
Set a sync callback function that must return a string that allows rewriting urls. 
```

What will happen if the user doesn't return a string?
Bad crash? - we should document it or define a behaviour

Reply:
Yeh, this is a good point. I also just realised that type definitions are missing so will add those in to give a level of protection for typescript users. 😅

We could guard against invalid return values (e.g. rewritten url needs to be of type string). 

How do you handle similar error cases in Fastify? 

Let the process crash or silently fail and continue?

Reply:
if we think this like a hook I would manage it returning 500 error to the single request since we do not crash anymore in v3

@mcollina do you agree?

Reply:
I think we should destroy the req/res with an error, which in turn should route this to the `'clientError'` event. This is really a low-level config.

I would recommend anybody using this to _rethink_ their design.

Reply:
> I would recommend anybody using this to _rethink_ their design.

This is another case where they should be using a proper reverse proxy like HAProxy in front of their Node server.